### PR TITLE
feat(ecommerce): map track events to Braze recommended ecommerce events

### PR DESCRIPTION
**Pull request by vgupta98**
**Fixes** # (N/A)

## Description
Adds support for Braze's recommended ecommerce events (`ecommerce.*`) in the device-mode integration, behind a new opt-in destination flag `useRecommendedEcommerceEvents` (default **off**). When the flag is off, behaviour is unchanged. When on, RudderStack ecommerce `track` events are mapped to the corresponding Braze recommended events via `logCustomEvent`, taking a hard cutover ahead of the legacy `Order Completed → logPurchase` path. All stateless mapping logic is extracted into a new `Utils` class, keeping `BrazeIntegrationFactory` focused on lifecycle and the SDK call.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Implementation Details
- **New opt-in flag** `useRecommendedEcommerceEvents` parsed from the destination config in the constructor (default `false`). Gating is applied only when `connectionMode == device`.
- **Event mapping** (case-insensitive) from RudderStack ecommerce events to Braze recommended events:
  - `Product Viewed → ecommerce.product_viewed`
  - `Product Added → ecommerce.cart_updated` (action `add`)
  - `Product Removed → ecommerce.cart_updated` (action `remove`)
  - `Checkout Started → ecommerce.checkout_started`
  - `Order Completed → ecommerce.order_placed`
  - `Order Refunded → ecommerce.order_refunded`
  - `Order Cancelled → ecommerce.order_cancelled`
  - Events with no recommended-event counterpart (e.g. Product Clicked, Cart Viewed, Install Attributed) fall through and keep their existing behaviour.
- **Hard cutover** for `Order Completed`: when the flag is on, it emits `ecommerce.order_placed` (one event per order) instead of the legacy per-product `logPurchase`.
- **Property mapping** uses RudderStack's own SDK constants (`ECommerceEvents` / `ECommerceParamNames`) for event and param names wherever the SDK exposes them; the few source keys without a constant (e.g. `sku`, `name`, `variant`, `image_url`, `url`, `value`, `tax`, `shipping`, `cancel_reason`) are kept as clearly-commented local literals.
- **Fallback chains** for resilient mapping: `total → revenue → value`, `product_id → sku`, `variant → sku → product_id`, `checkout_id → order_id`.
- **Metadata pass-through**: any source key not consumed by a top-level/product mapping is routed into `metadata` / `products[].metadata`, never left as an unexpected top-level property (which Braze rejects).
- **Send-anyway validation**: missing Braze-required fields are logged via `logger.warn` but the event is still sent (no drops); `source` is hardcoded per platform (`android`).
- **Refactor**: extracted the enum, all mapping constants, the lookup table, per-event builders, and helpers into a new stateless `Utils` class; the factory now delegates via `Utils.resolveEcommerceEvent(...)` and `Utils.buildEcommerceProperties(...)` and performs the single `braze.logCustomEvent` call.
- **No SDK bump**: nested `products[]`/`metadata` custom-event properties are supported on the current Braze SDK pin.

## Checklist:
- [ ] Version upgraded (project, README, gradle, podspec etc)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

## How to test?
1. Configure a Braze destination in device mode with `useRecommendedEcommerceEvents` enabled.
2. Send the supported ecommerce `track` events (e.g. `Product Viewed`, `Product Added`, `Order Completed`) with representative properties.
3. Verify in Braze that the corresponding `ecommerce.*` custom events are received with the expected mapped properties, `products[]`/`metadata` nesting, and `source = android`.
4. Toggle the flag off and confirm legacy behaviour is unchanged (`Order Completed → logPurchase`).

## Breaking Changes
None. The feature is gated behind a new flag that defaults to off; existing behaviour is preserved when the flag is disabled.

### CODE REVIEW

File: braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java [hunk 1]
```diff
@@ -332,6 +340,21 @@ private void processTrackEvent(RudderMessage element) {
...
+            // counterpart (Product Clicked, Cart Viewed, Install Attributed, etc.) fall through
+            // and keep their existing behaviour.
+            if (useRecommendedEcommerceEvents) {
+                Utils.EcommerceEvent ecommerceEvent = Utils.resolveEcommerceEvent(event);
+                if (ecommerceEvent != null) {
+                    Map<String, Object> brazeProperties = Utils.buildEcommerceProperties(ecommerceEvent, eventProperties);
+                    RudderLogger.logDebug(String.format(
+                            "Braze logCustomEvent for recommended ecommerce event %s with properties %% %s",
+                            ecommerceEvent.getBrazeEvent(), brazeProperties.toString()));
+                    this.braze.logCustomEvent(ecommerceEvent.getBrazeEvent(), new BrazeProperties(brazeProperties));
```
Comment (dciccale) [inline braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java]: `BrazeProperties(Map)` in the pinned Braze SDK drops `List` values, so `products`, `discounts`, and `type` can disappear before dispatch. Can we convert these arrays to `JSONArray` before calling `logCustomEvent`?

**Review by dciccale — COMMENTED** [review]
check below

File: braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+        if (eventName == null) {
+            return null;
+        }
+        return ECOMMERCE_EVENT_MAPPING.get(eventName.trim().toLowerCase(Locale.US));
+    }
+
+    // Builds the Braze custom-event property map for a recommended ecommerce event.
+    // Send-anyway posture (D5): build from an empty map when properties are absent so the event is
+    // still logged (with required-field warnings) rather than dropped.
+    static Map<String, Object> buildEcommerceProperties(EcommerceEvent ecommerceEvent, Map<String, Object> properties) {
```
Comment (dciccale) [inline braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java:211]: Can we add focused unit tests for the new mapper? At minimum I’d cover `Order Completed -> ecommerce.order_placed`, metadata pass-through, numeric coercion, and the final `BrazeProperties` payload for array fields like `products`.

**Review by dciccale — COMMENTED** [review]
check below

**Review by dciccale — CHANGES_REQUESTED** [review]

File: braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java (hunk 1 above)
Comment (vgupta98) [reply braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java]: Good catch. Fixed in `67bedde` — now converting the map to a JSONObject (lists → JSONArray) before `BrazeProperties`. Verified on-device: products/discounts/type make it to the wire now.

File: braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java (hunk 2 above)
Comment (vgupta98) [reply braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java:211]: Added in `23900a7`. Covers order_placed mapping, metadata pass-through, numeric coercion, and asserts the final payload keeps products/discounts/type as JSONArrays (the array case you flagged).

File: braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+            return null;
+        }
+        List<?> rsProducts = (List<?>) raw;
+        List<Map<String, Object>> products = new ArrayList<>();
+        for (Object item : rsProducts) {
+            if (item instanceof Map) {
+                products.add(buildProduct((Map<String, Object>) item));
+            }
+        }
+        return products.isEmpty() ? null : products;
```
[the remark this answers is not in the document]
Comment (vgupta98) [reply braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java]: Fixed — `buildProducts` now skips empty product maps, so an all-empty array is treated as missing (omitted + warning).

File: braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java [hunk 4]
```diff
@@ -332,6 +340,21 @@ private void processTrackEvent(RudderMessage element) {
...
+            // legacy Order Completed / logCustomEvent path. Events with no recommended-event
+            // counterpart (Product Clicked, Cart Viewed, Install Attributed, etc.) fall through
+            // and keep their existing behaviour.
+            if (useRecommendedEcommerceEvents) {
+                Utils.EcommerceEvent ecommerceEvent = Utils.resolveEcommerceEvent(event);
+                if (ecommerceEvent != null) {
+                    Map<String, Object> brazeProperties = Utils.buildEcommerceProperties(ecommerceEvent, eventProperties);
+                    RudderLogger.logDebug(String.format(
+                            "Braze logCustomEvent for recommended ecommerce event %s with properties %% %s",
+                            ecommerceEvent.getBrazeEvent(), brazeProperties.toString()));
```
[the remark this answers is not in the document]
Comment (vgupta98) [reply braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java]: Fixed — dropped the stray `%%`; that line now logs the JSON payload directly.

File: braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+            return null;
+        }
+        return ECOMMERCE_EVENT_MAPPING.get(eventName.trim().toLowerCase(Locale.US));
+    }
+
+    // Builds the Braze custom-event property map for a recommended ecommerce event.
+    // Send-anyway posture (D5): build from an empty map when properties are absent so the event is
+    // still logged (with required-field warnings) rather than dropped.
+    static Map<String, Object> buildEcommerceProperties(EcommerceEvent ecommerceEvent, Map<String, Object> properties) {
+        Map<String, Object> props = (properties != null) ? properties : new HashMap<>();
```
[the remark this answers is not in the document]
Comment (vgupta98) [reply braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java:212]: Added unit tests in `23900a7` covering these transforms — mapping, metadata pass-through, coercion, plus array preservation in the final payload.

File: braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+                    Long parsed = parseLongOrNull((String) value);
+                    return parsed != null ? parsed : value;
+                }
+                return value;
+            case STRING_ARRAY:
+            case ARRAY:
+            default:
+                return value;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (vgupta98) [reply braze/src/main/java/com/rudderstack/android/integration/braze/Utils.java:680]: Intentional — we don't synthesize an array the source didn't send. Un-coercible array fields are passed through as-is and surfaced via the type-mismatch warning rather than guessed at.

**Review by dciccale — APPROVED** [review]

**Review by dciccale — APPROVED** [review]

**Review by SKannaniOS — APPROVED** [review]
LGTM
